src/WorkItems/Logging/MetricsLogValues.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.Extensions.Logging;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.WorkItems.Logging
+{
+    /// <summary>
+    /// MetricsLogValues is variation of FormattedLogValues that accepts a dictionary to store customDimensions Application Insights.
+    /// https://github.com/aspnet/Logging/blob/master/src/Microsoft.Extensions.Logging.Abstractions/Internal/FormattedLogValues.cs
+    /// </summary>
+    internal readonly struct MetricsLogValues : IReadOnlyList<KeyValuePair<string, object>>, IEnumerable<KeyValuePair<string, object>>, IEnumerable, IReadOnlyCollection<KeyValuePair<string, object>>
+    {
+        private readonly List<KeyValuePair<string, object>> _values;
+        private readonly string _originalMessage;
+        private readonly EventId _eventId;
+
+        public KeyValuePair<string, object> this[int index]
+        {
+            get
+            {
+                if (index < 0 || index >= Count)
+                {
+                    throw new IndexOutOfRangeException("index");
```

Comment:
>IndexOutOfRangeException [](start = 30, length = 24)

Aren't you just recreating existing index behavior here?

Reply:
Yeah, it's from the original implementation. When I removed some of the logic, it was reduced to this. I'll remove the if block.

---
In reply to: [396828009](https://github.com/microsoft/sarif-sdk/pull/1820#discussion_r396828009) [](ancestors = 396828009)